File: all-submissions/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int[] answer = new int[2];
        for(int i=0; i<nums.length; i++) {
            int firstNum = nums[i];
            int secondNum = target - nums[i];
            if (map.containsKey(secondNum)) {
                int secondIndex = map.get(secondNum);
                if (secondIndex != i) {
                    answer[0] = i;
                    answer[1] = map.get(secondNum);
                    break;
                }
            }
            map.put(nums[i], i);
        } 
        return answer;
    }
}

File: all-submissions/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        dict = {}
        for i in range(len(nums)):
            remain = target-nums[i]
            if remain in dict.keys():
	            return [dict[remain], i]
            dict[nums[i]] = i
        return [-1,-1]



